datamodel/samples.d/generate_timestampdefinitions.py

```diff
@@ -0,0 +1,1188 @@
+        "Port Departure Execution",
+        'jit1_1',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=ACT,
+    )
+
+    #ATS Towage UC 85 + ATC Towage UC 86
+    xty_service_timestamps(
+        ACT,
+        ['TOWG'],
+        ['OUTB', 'SHIF'],
+        "Port Departure Execution",
+        'jit1_1',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=ACT,
+    )
+
+    #SOSP UC 88
+    generate_special_timestamp(
+        'SOSP',
+        PUBLISHER_PATTERN_CA2ATH,
+        'DEPA',
+        NULL_VALUE,
+        'OUTB',
+        "Port Departure Execution",
+        'jit1_1',
+        NULL_VALUE,
+        need_vessel_position=False,
+        need_event_location=False,
+    )
+
+    # Anchorage UC 89 - UC 96
+    generic_xty_timestamps(
+        PUBLISHER_PATTERN_CA2ATH,
+        EST_PLN_REQ_ACT,
+        ['ARRI', 'DEPA'],
+        ['ANCH'],
+        [NULL_VALUE],
+        "Other Services - Anchorage Planning And Execution",
+        'jit1_2',
+        include_phase_in_name=True,
+        need_event_location_for=ALL_EVENT_CLASSIFIER_CODES,
+        need_vessel_position_for=ALL_EVENT_CLASSIFIER_CODES
+    )
+
+    #ATS Anchorage OPS UC 97
+    generate_special_timestamp(
+        'ATS Anchorage Ops',
+        PUBLISHER_PATTERN_CA2ATH,
+        'ARRI',
+        'ANCH',
+        NULL_VALUE,
+        "Other Services - Anchorage Planning And Execution",
+        'jit1_2',
+        'ANCO',
+        need_vessel_position=True, #FIXME need to support optional
+        need_event_location=True,
+    )
+
+    #ATC Anchorage OPS UC 98
+    generate_special_timestamp(
+        'ATC Anchorage Ops',
+        PUBLISHER_PATTERN_CA2ATH,
+        'DEPA',
+        'ANCH',
+        NULL_VALUE,
+        "Other Services - Anchorage Planning And Execution",
+        'jit1_2',
+        'ANCO',
+        need_vessel_position=True, #FIXME need to support optional
+        need_event_location=True,
+    )
+
+    #Sludge UC 99 106
+    generic_xty_timestamps(
+        as_publisher_patterns(['CA'], ['SVP']),
+        EST_PLN_REQ_ACT,
+        ['STRT', 'CMPL'],
+        ['BRTH'],
+        [NULL_VALUE],
+        "Other Services - Sludge Planning And Execution",
+        'jit1_2',
+        ['SLUG'],
+        need_vessel_position_for=EST_PLN_REQ_ACT,
+        need_event_location_for=EST_PLN_REQ_ACT,
+    )
+
+    #Shore power UC 107 + 108
+    xty_service_timestamps(
+        ACT,
+        ['SHPW'],
+        ['ALGS'],
+        "Other Services - Shore Power Execution",
+        'jit1_2',
+        is_cancelable=True,
+        need_event_location_for=ACT,
+    )
+
+    #Cancel port call UC 109
+    generate_special_timestamp(
+        'Cancel Port Call',
+        PUBLISHER_PATTERN_CA2ATH,
+        'CANC',
+        NULL_VALUE,
+        NULL_VALUE,
+        "OMIT Port Call Or Cancel A Service",
+        'jit1_2',
+        need_vessel_position=True, #FIXME need to support optional
+    )
```

Comment:
I am not sure this is relevant.  Ideally, we would have generated all the relevant cancels as a part of the services (`is_cancelable`).  If this is a special case variant not covered by others, then the name is wrong (you cannot cancel a port call - that is the next use case and is called Omit) and should probably be something on the line of `Cancel Service ...`.

Reply:
removed this section